### PR TITLE
Add the python api for vineyard llm kv cache

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

Fixes https://github.com/v6d-io/v6d/issues/1768

### CODE REVIEW

**Comment by dashanji** [comment]
Wait for refine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +50,7 @@ option(BUILD_VINEYARD_CLIENT "Build vineyard's client" ON)
 option(BUILD_VINEYARD_CLIENT_VERBOSE "Build vineyard's client with the most verbose logs" OFF)
 option(BUILD_VINEYARD_PYTHON_BINDINGS "Build vineyard's python bindings" ON)
 option(BUILD_VINEYARD_PYPI_PACKAGES "Build vineyard's python bindings" OFF)
+option(BUILD_VINEYARD_LLM_KV_CACHE_PYTHON_BINDINGS "Build vineyard llm kv cache's python bindings" ON)
```
Comment (sighingnow) [inline CMakeLists.txt:53]: No need for this extra option.

`if (BUILD_VINEYARD_PYTHON_BINDINGS AND BUILD_VINEYARD_LLM_CACHE)` would be enough.

File: python/vineyard/llm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            v_tensor_numpy = np.frombuffer(
+                v_state.data, dtype=self.numpy_dtype
+            ).reshape(self.torch_size)
+            k_tensor = torch.from_numpy(k_tensor_numpy)
+            v_tensor = torch.from_numpy(v_tensor_numpy)
+            kv_cache_list.append((k_tensor, v_tensor))
+        return kv_cache_list
+
+    def clear(self):
+        self.kv_cache_manager.close()
```
Comment (sighingnow) [inline python/vineyard/llm/__init__.py:119]: `clear()` shouldn't be implemented as `close()`.

You may want `__del__` as the destructor.

**Review by sighingnow — APPROVED** [review]

File: python/vineyard/llm/__init__.py (hunk 2 above)
Comment (dashanji) [reply python/vineyard/llm/__init__.py:119]: Make sense to me. Thanks.

File: CMakeLists.txt (hunk 1 above)
Comment (dashanji) [reply CMakeLists.txt]: Done.

**Pull request merged**